velox/functions/prestosql/Arithmetic.h

```diff
@@ -69,6 +69,32 @@ struct MultiplyFunction {
   }
 };
 
+template <typename T>
+struct IntervalMultiplyFunction {
+  double sanitizeInput(double d) {
+    if UNLIKELY (std::isnan(d)) {
+      return 0;
+    }
+    return d;
+  }
+
+  template <typename T1, typename T2>
+  FOLLY_ALWAYS_INLINE void call(int64_t& result, const T1 a, const T2 b) {
+    double resultDouble;
+    if constexpr (std::is_same_v<T1, double>) {
+      resultDouble = sanitizeInput(a) * b;
+    } else {
+      resultDouble = sanitizeInput(b) * a;
+    }
+    if LIKELY (std::isfinite(resultDouble)) {
```

Comment:
curious does isfinite guard against overflow etc ?

Reply:
Good catch. I added the handling of overflow and the test cases.